internal/domain/pdfgen/model.go

```diff
@@ -0,0 +1,77 @@
+package pdfgen
+
+import (
+	"time"
+
+	"github.com/shopspring/decimal"
+)
+
+// InvoiceData represents the data model for invoice PDF generation
+type InvoiceData struct {
+	ID              string          `json:"ID"`
+	InvoiceNumber   string          `json:"InvoiceNumber"`
+	CustomerID      string          `json:"CustomerID"`
+	SubscriptionID  string          `json:"SubscriptionID,omitempty"`
+	InvoiceType     string          `json:"InvoiceType"`
+	InvoiceStatus   string          `json:"InvoiceStatus"`
+	PaymentStatus   string          `json:"PaymentStatus"`
+	Currency        string          `json:"Currency"`
+	AmountDue       decimal.Decimal `json:"AmountDue"`
+	AmountPaid      decimal.Decimal `json:"AmountPaid"`
+	AmountRemaining decimal.Decimal `json:"AmountRemaining"`
+	Description     string          `json:"Description,omitempty"`
+	DueDate         time.Time       `json:"DueDate,omitempty"`
+	PaidAt          *time.Time      `json:"PaidAt,omitempty"`
+	VoidedAt        *time.Time      `json:"VoidedAt,omitempty"`
+	FinalizedAt     *time.Time      `json:"FinalizedAt,omitempty"`
```

Comment:
Why not use normal json tag convention we have been following ex voided_at, paid_at, etc?